litestar/plugins/problem_details.py

```diff
@@ -0,0 +1,140 @@
+"""Plugin for converting exceptions into a problem details response."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING, Any, Callable, Mapping, TypeVar
+
+from typing_extensions import TypeAlias
+
+from litestar.exceptions.http_exceptions import HTTPException
+from litestar.plugins.base import InitPluginProtocol
+from litestar.response.base import Response
+
+if TYPE_CHECKING:
+    from litestar.config.app import AppConfig
+    from litestar.connection.request import Request
+    from litestar.types.callable_types import ExceptionHandler, ExceptionT
+
+ProblemDetailsExceptionT = TypeVar("ProblemDetailsExceptionT", bound="ProblemDetailsException")
+ProblemDetailsExceptionHandlerType: TypeAlias = "Callable[[Request, ProblemDetailsExceptionT], Response]"
+ExceptionToProblemDetailMapType: TypeAlias = (
+    "Mapping[type[ExceptionT], Callable[[ExceptionT], ProblemDetailsExceptionT]]"
+)
+
+
+def _problem_details_exception_handler(request: Request[Any, Any, Any], exc: ProblemDetailsException) -> Response[Any]:
+    return exc.to_response(request)
+
+
+def _create_exception_handler(
+    exc_to_problem_details_exc_fn: Callable[[ExceptionT], ProblemDetailsException], exc_type: type[ExceptionT]
+) -> ExceptionHandler:
+    def _exception_handler(req: Request, exc: exc_type) -> Response:  # type: ignore[valid-type]
+        problem_details_exc = exc_to_problem_details_exc_fn(exc)
+
+        return problem_details_exc.to_response(req)
+
+    return _exception_handler
+
+
+def _http_exception_to_problem_detail_exception(exc: HTTPException) -> ProblemDetailsException:
+    return ProblemDetailsException(status_code=exc.status_code, detail=exc.detail, extra=exc.extra, headers=exc.headers)
```

Comment:
Perhaps `HTTPException.detail` should rather map to to the `title` field? The `detail` field in `HTTPException` typically contains just a terse HTTP status name, e.g. `Bad Request`.